packages/shared/src/errors.ts

```diff
@@ -21,7 +21,15 @@ import { bip39Words } from '@/bip39'
 const bip39WordsString = bip39Words.join('|')
 
 export const getHumanReadableError = (error: unknown, defaultErrorMsg: string) =>
-  typeof error?.toString === 'function' ? error.toString().replace('Error: [API Error] - ', '') : defaultErrorMsg
+  defaultErrorMsg +
+  (errorHasMessageProp(error)
+    ? ` - ${error.message}`
+    : typeof error?.toString === 'function'
+      ? ` - ${error.toString().replace('Error: [API Error] - ', '')}`
+      : '')
 
 export const cleanExceptionMessage = (error: unknown) =>
   getHumanReadableError(error, '').replace(new RegExp(`\\b(${bip39WordsString})\\b`, 'g'), '[...]')
+
+const errorHasMessageProp = (error: unknown): error is { message: string } =>
+  (error as { message: string })?.message !== undefined
```

Comment:
Nit: I used the `in` notation extensively in my tanstack branch and I like it, it's easily readable. 
Here we could do: 

```ts
const errorHasMessageProp = (error: unknown): error is { message: string } =>
  'message' in (error as { message?: string })
```